### DISCUSSION TITLE
Screen capture from OBS - FS Kit WIP??

### DISCUSSION THREAD

**Discussion by androooo300** [springbok/MLM2PRO-GSPro-Connector#75] [Q&A]
Is it possible to get the ROI's from OBS? 

I have tried to link the device to OBS, but it won't find the window.

Only reason I ask this, is I'm experimenting with a Fullswing Kit Launch Monitor. I can't mirror the ipad through airplay while being connected to the launch monitor via wifi.

**Comment by springbok** [comment]
Did you update the window title in the device to whatever the title of the OBS window is?

**Reply by androooo300** [reply]
Yes... I have now got it working. Have been able to get to ROI's to work, however "1's" get read as "7's" due to the font used. Is there any way of fixing that?

I hit a wedge shot, and it read almost perfectly :)
7 iron, because it launched at 17*, got read as 77

**Comment by springbok** [comment]
The OCR needs to be trained to recognize the font. Paul can help create these files from highly quality images.

**Reply by androooo300** [reply]
@wonder99 What kind of screenshots do you need for the OCR training?

**Reply by androooo300** [reply]
@wonder99 ![image](https://github.com/springbok/MLM2PRO-GSPro-Connector/assets/156437432/23b778f5-d77d-40bd-951f-2cea4a9971ba)

Does this help at all?

**Reply by springbok** [reply]
Thanks for the image, Paul is away on a golf trip at the moment, I'm sure he will have a look once he is back. He would need a number of images that have all possible digits 0-9, signs and characters.

**Reply by wonder99** [reply]
Yes, I'm back on Monday. I'll catch up on this when I return.

**Reply by androooo300** [reply]
I will be at work on Monday, with full access to the FSKit. If you need any other screen shots, or need me to test anything, just let me know.

**Reply by wonder99** [reply]
Can you send me uncompressed PNG images with examples of all digits and characters?  plus, minus, degree, etc,  I only see the one snapshot here.
I'm a little worried about the comma in the spin number--we may need a code change to strip that out as well.
When snapshotting the images, make sure they are taken when the FSKit window is as big as possible (e.g. full screen)

**Reply by androooo300** [reply]
I am working tonight, so I will take a bunch of screenshots of different shots. Do you want screenshots of OBS, as that is how it is being brought into the connector, or of the ROI screen?

**Comment by wonder99** [comment]
I'm not familiar with obs, but I need the screen shots from the same
computer that is running the connector.  So I'm guessing I need the obs
view of the ROIs.
Paul

**Comment by androooo300** [comment]
@wonder99  Hopefully these are ok. If there is anything else, just let me know

![FSKit1](https://github.com/springbok/MLM2PRO-GSPro-Connector/assets/156437432/c172510a-22c0-43fe-aa5d-926521594dfb)
![FSKit2](https://github.com/springbok/MLM2PRO-GSPro-Connector/assets/156437432/bd3d5157-f694-4f63-b393-4501cab58174)
![FSKit3](https://github.com/springbok/MLM2PRO-GSPro-Connector/assets/156437432/2f9f0ad2-d7ea-4469-a22b-efbf2adbe3fb)
![FSKit4](https://github.com/springbok/MLM2PRO-GSPro-Connector/assets/156437432/8feec8d5-e068-402e-be35-686e3a49063b)
![FSKit5](https://github.com/springbok/MLM2PRO-GSPro-Connector/assets/156437432/ebbc567b-7a85-41db-82bf-50b06a41e07b)
![FSKit6](https://github.com/springbok/MLM2PRO-GSPro-Connector/assets/156437432/20534226-1a5f-4a3c-afbc-2bcbdfdde8dd)

**Comment by wonder99** [comment]
Looks great, thanks

**Comment by wonder99** [comment]
[fskit.zip](https://github.com/springbok/MLM2PRO-GSPro-Connector/files/14363631/fskit.zip)
I'm attaching here a zipped version of the training file.  For now, you can overwrite your train.traineddata file with this and select MLM2PRO as your monitor type in the connector.  Later, Etienne will probably add an FSKit monitor option, and the existing filename will be recognized.
This OCR should work very well, the source images are very clear.  Note that commas are passed through, so Etienne, not sure if you already strip those out, but that will be required.

**Reply by androooo300** [reply]
Many thanks Paul. I will have a chance to test it on Friday. I will report back.

Love your work!

**Reply by springbok** [reply]
Let us know how it goes, and I will add a new LM option  to settings.

**Reply by androooo300** [reply]
Having issues with the spin rates... guessing it's reading the comma. Aside from that, all numbers are read correctly.

![error](https://github.com/springbok/MLM2PRO-GSPro-Connector/assets/156437432/a99f34a8-28ae-4b36-a5ee-b3ac15395eb8)

**Reply by androooo300** [reply]
Correction... it is reading correctly... some of the time. I think it's because of the delay in the spinrate being shown.
![error2](https://github.com/springbok/MLM2PRO-GSPro-Connector/assets/156437432/c86872c7-2f3d-4ba0-8ef9-b96cdb6b2565)
https://github.com/springbok/MLM2PRO-GSPro-Connector/assets/156437432/5859afe2-c359-4866-ae8c-5e2c43db2630 video of data capture

**Reply by springbok** [reply]
Here is a link to v1.02.02, set the LM in settings, save, restart connector. I've not tested it so let me know if you strike any problems.

https://1drv.ms/u/s!Agx7aYfjrZJ8qDyFkZ4kpZZdonxX?e=RsIH1s

**Reply by androooo300** [reply]
Thanks @springbok... Seemed to work great in a quick test today.
 
I will do some further testing when I get a chance, but seemed very stable and reliable. Many thanks to you and @wonder99 for the help getting this even a chance at working.

**Comment by springbok** [comment]
I'll add it as a new LM and strip the comma, I'll share a link to the update when its ready.

**Comment by androooo300** [comment]
@springbok More than happy with the version you gave me to test. It works as well as possible, and can be released.

It should be noted, that due to limitations with the FSKit at present, shots under approx 30m will not be picked up. I have asked FS if there is a chance if we can have a "short mode" activated in the FSKit app.

When I get a chance, I will try to write a readme on how to set up for the FSKit.

Many thanks to you and @wonder99 for all your work on this.

**Reply by springbok** [reply]
No worries, great to hear its working for you. It would be great if you can do the readme in may help other users.

**Comment by kay7711226** [comment]
Hello, I am interested in how this was done. I have the FSK launch monitor, however the only interface APP is via iPad and as user mentioned it cannot mirror while connected to launch monitor. So how was it done? I see OBS was used and would like to know how. See you mentioned the readme being updated to help others but can't find it either. Thanks for the great work.
